bindings/rust/build.rs

```diff
@@ -1,170 +1,58 @@
-#[cfg(feature = "build_unicorn_cmake")]
-use bytes::Buf;
-#[cfg(feature = "build_unicorn_cmake")]
-use flate2::read::GzDecoder;
 #[cfg(feature = "use_system_unicorn")]
 use pkg_config;
 #[cfg(feature = "build_unicorn_cmake")]
-use reqwest::header::USER_AGENT;
-#[cfg(feature = "build_unicorn_cmake")]
-use std::path::{Path, PathBuf};
+use std::env;
 #[cfg(feature = "build_unicorn_cmake")]
-use std::{env, process::Command};
-#[cfg(feature = "build_unicorn_cmake")]
-use tar::Archive;
-
-#[cfg(feature = "build_unicorn_cmake")]
-fn find_unicorn(unicorn_dir: &Path) -> Option<PathBuf> {
-    for entry in std::fs::read_dir(unicorn_dir).ok()? {
-        let entry = entry.unwrap();
-        let path = entry.path();
-
-        if path.is_dir() && path.file_name()?.to_str()?.contains("unicorn") {
-            return Some(path);
-        }
+use std::path::Path;
+
+#[cfg(all(feature = "build_unicorn_cmake", target_os = "windows"))]
+fn setup_env_windows() {
+    let devenv_path = cc::windows_registry::find_tool(
+        env::var("TARGET").unwrap().as_str(), "devenv").unwrap();
+    let devenv_dir = devenv_path.path().parent().unwrap();
+
+    let cmake_path = devenv_dir.join(r"CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe");
+    let ninja_path = devenv_dir.join(r"CommonExtensions\Microsoft\CMake\Ninja\ninja.exe");
+    if !cmake_path.is_file() {
+        panic!("missing cmake");
     }
-
-    None
-}
-
-#[cfg(feature = "build_unicorn_cmake")]
-fn out_dir() -> PathBuf {
-    let out_dir = env::var("OUT_DIR").unwrap();
-    Path::new(&out_dir).to_path_buf()
-}
-
-#[cfg(feature = "build_unicorn_cmake")]
-fn download_unicorn() -> PathBuf {
-    // https://docs.github.com/en/rest/reference/repos#download-a-repository-archive-tar
-    let pkg_version;
-    if let Ok(unicorn_version) = env::var("UNICORN_VERSION") {
-        pkg_version = unicorn_version;
-    } else {
-        pkg_version = env::var("CARGO_PKG_VERSION").unwrap();
+    if !ninja_path.is_file() {
+        panic!("missing ninja");
     }
-    let out_dir = out_dir();
-    let client = reqwest::blocking::Client::new();
-    let resp = client
-        .get(format!(
-            "https://api.github.com/repos/unicorn-engine/unicorn/tarball/{}",
-            pkg_version
-        ))
-        .header(USER_AGENT, "unicorn-engine-rust-bindings")
-        .send()
-        .unwrap()
-        .bytes()
-        .unwrap();
-    let tar = GzDecoder::new(resp.reader());
 
-    let mut archive = Archive::new(tar);
-    archive.unpack(&out_dir).unwrap();
-
-    find_unicorn(&out_dir).unwrap()
+    if let Some(path) = env::var_os("PATH") {
```

Comment:
I just saw you are patching `PATH` env and to be honest I don't think it's a clean and good solution...

Reply:
we have 2 choices when modifying `PATH`: append or prepend. I've chosen append here. lmk if it should be prepend.

Reply:
I suggest restoring the `PATH` variable after everything is done.

Reply:
bit confused what you're trying to prevent? cargo compiles and runs build.rs. i.e. unicorn's build.rs will run as it's own process - any env vars it modifies will not be persisted to the parent process or external system. see https://doc.rust-lang.org/std/env/fn.set_var.html . persisting env vars outside your process is OS/shell specific.

btw, there's also this in rust docs (https://doc.rust-lang.org/cargo/reference/build-scripts.html):
> The build script will be rebuilt if any of its source files or dependencies change.
> By default, Cargo will re-run the build script if any of the files in the package changes. Typically it is best to use the rerun-if commands, described in the [change detection](https://doc.rust-lang.org/cargo/reference/build-scripts.html#change-detection) section below, to narrow the focus of what triggers a build script to run again.
see also https://github.com/rust-lang/cargo/issues/2261#issuecomment-170461557

So the `cargo:rerun-if-changed` lines which unicorn's build.rs script has are not required.